lib/ansible/modules/cloud/vultr/vultr_server.py

```diff
@@ -517,7 +540,7 @@ def _create_server(self, server=None):
                 data=data
             )
             server = self._wait_for_state(key='status', state='active')
-            server = self._wait_for_state(state='running')
+            server = self._wait_for_state(state='running', timeout=3600 if snapshot_restore else 60)
```

Comment:
stopping a play for one hour very long, i am evaluation other options how to handle this use case. do you know, if we can find out that a restoring is currently going on? I mean we should probably this case anyway and not trying to start an existing VM currently restoring manually triggered or a user may hit ctrl-c and restart the playbook.

Another option is that we add another flag `wait_for_snapshot` with default=yes to give users a choice not to wait for it.

any thoughts?

Reply:
An hour is the worst-case scenario. Most snapshots restore in less time than that, but Vultr doesn't guarantee anything less than an hour.

If a user does not want to wait for the snapshot restore to complete, Ansible already has a built-in solution: use [async](https://docs.ansible.com/ansible/latest/user_guide/playbooks_async.html).

One thing the module does lack is detecting if a server's `server_state` is `locked.` Technically, no actions can be performed on a locked server. Maybe a future change to make the module more resilient to Ctrl+C replays might be waiting for the `server_state` to reach `ok` before executing any actions.

Reply:
OK, I updated the PR to wait for a locked server when trying to `update_server`. I tested interrupting and restarting a playbook and it correctly waits for the snapshot restore to finish before continuing with the update call. Let me know what you think.

Reply:
thanks for the valuable input.